### PR TITLE
* Improve style of top nav search input

### LINKED ISSUE #1387: Little (x) Button To Quickly Clear Search Bar Of Text
**Is your feature request related to a usage problem (not a bug)? Please describe.**
Usage. Less clicks means better UI. I've seen similar implementation where clicking the X button (whatever color such as red, black or white) will clear the search text and select the search bar.

One button and you're ready to type in your new search term.

Current way is to click the bar then use Backspace but this can feel annoying after a while compared to what can be done with one click of an X button.

**Describe the solution you'd like to see implemented**
Add an X button that will clear and select the search bar in one action/click. 

Best spot to put it for the current UI would be on the left side of the bar instead of next to the search arrow and filter buttons as that would start to make things cluttered there and would prevent accidental clicks that clear your current search.

--

Possible improvements would to be to add a couple hotkeys. One to clear text function while within the search bar or even while the FreeTube window is selected. The other to quickly navigate from anywhere to the search bar.

Possible clear function default hotkeys: [Ctrl]+[Backspace] or [Ctrl]+[Delete]

Possible hotkey for navigating to the search bar is: [Ctrl]+[Shift]+[F]

(I know hotkeys are not implement yet. These are future possibilities.)

**Describe alternatives you've considered**
Nothing that's as simple to implement or use. Anything more would be clunky.

**Screenshots**
Searchbar Mockup

**Additional context**
I realize that #1392 was created as an improvement to mine. I'm no UI designer and only used MS Paint to create my mockup. Feel free to merge the issues if you feel it's applicable while adding anything that's not in the other together.

### PR DESCRIPTION
**Pull request by PikachuEXE**
**Pull Request Type**
Please select what type of pull request this is:
- [x] Feature Implementation

**Related issue**
Closes #1387

**Description**
Update top nav search to add clear search input text button
Also update the padding of input text box so that input text won't appear under the "arrow-right" button

**Screenshots (if appropriate)**
https://user-images.githubusercontent.com/1018543/120764623-ebd83900-c54a-11eb-95da-a50112ef01a7.mp4
image

**Testing (for code that is not small enough to be easily understandable)**
- Enter text very long into input text box
- Press the new button

**Desktop (please complete the following information):**
 - OS: MacOS
 - OS Version: 10.15.7
 - FreeTube version: 13.0.0 (latest on `development` actually)

**Additional context**

### CODE REVIEW

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
By the way u have to add the word "closes" before the issue number to link the issue. After this PR is merged the issue will close automatically instead of staying open and clutter the backlog.

**Comment by PikachuEXE** [comment]
I know it already
I do want a feature to auto post a comment on linked issue after PR merged instead of closing it
Sometimes an issue is not meant to be fixed completely by one PR
I will add "closes" since there is no such feature yet -,-"

**Comment by PikachuEXE** [comment]
image
Oops

**Comment by PikachuEXE** [comment]
Issue shown in my previous comment fixed
I still got no idea how to test the look in RTL

**Comment by ChunkyProgrammer** [comment]
I think it'd be best if this gets merged first then i can update it to be rtl compatible in my pr

**Comment by peepopoggers** [comment]
@PikachuEXE has the placement changed?

File: src/renderer/components/ft-input/ft-input.vue
Comment (jasonhenriquez) [inline src/renderer/components/ft-input/ft-input.vue:28]:
```suggestion
    <font-awesome-icon
      v-if="showClearTextButton"
      icon="backspace"
      class="clearInputTextButton"
      tabindex="0"
      role="button"
      title="$t('Clear Input')"
      @click="handleClearTextClick"
      @keydown.space.prevent="handleClearTextClick"
      @keydown.enter.prevent="handleClearTextClick"
     />
```
This is inaccessible. I suggest you implement these changes here and create a yaml string "Clear Input" as well to use here as the title.

**Review by jasonhenriquez — CHANGES_REQUESTED** [review]
I have not tested myself yet but can preliminarily see one accessibility flaw and a need for a new yaml string.

**Comment by PikachuEXE** [comment]
Fixed issue raised by @jasonhenriquez
Updated icon to [times-circle](https://fontawesome.com/v5.15/icons/times-circle?style=solid)


https://user-images.githubusercontent.com/1018543/131792332-e5648f58-9d21-4f70-87b3-f38b730f9aa8.mov

**Comment by jasonhenriquez** [comment]
Looks good accessibility-wise. But visually looks odd position-wise. It needs to be positioned with more space to its left.

**Comment by PikachuEXE** [comment]
Screenshot 2021-09-02 at 3 18 36 PM
Updated again
What do you think?

**Comment by jasonhenriquez** [comment]
I think it would be best if the icon was a bit more right such that its hover radius was touching the text like the arrow icon:
pr-hover

**Comment by PikachuEXE** [comment]
Done
Now using `2em` as `padding-left`, same as `padding-right`

Screenshot 2021-09-02 at 3 30 30 PM
Screenshot 2021-09-02 at 3 30 35 PM
Screenshot 2021-09-02 at 3 30 39 PM

**Review by jasonhenriquez — APPROVED** [review]

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Review by peepopoggers — CHANGES_REQUESTED** [review]
I think the button should dissapear when the search input is empty.

Also the placement of the button is strange, on the left of the -> would make more sense to me.

And also as I said before an icon with just a cross would look much better because the -> is not in a circle so it breaks the consistency.

**Comment by peepopoggers** [comment]
image
What I think would make more sense placement-wise.

Using an icon [like this](https://fontawesome.com/v5.15/icons/times?style=solid)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I disagree 

> Best spot to put it for the current UI would be on the left side of the bar instead of next to the search arrow and filter buttons as that would start to make things cluttered there and would prevent accidental clicks that clear your current search.

And i agree with this. You have to remember people that use smartphones also use FreeTube. U would definitely miss click if its next to each other.

**Comment by peepopoggers** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc
People that use phones will use the enter key on their keyboard most likely instead of the x button

**Comment by PikachuEXE** [comment]
@peepopoggers 
As a mobile app user, I can use the app with
1. Having the clear button on the right of text input box, **replacing** the search button (but I think there are people dislike removing existing buttons)
2. Having the clear button on the left of text input box (current)

But I cannot use the app with
1. Having the clear button on the left of search button (mis-clicking happens so often on mobile devices)

As a desktop app user, I can use the app with
1. Having the clear button on the left of text input box (current)

But I cannot use the app with
1. Having the clear button on the right of text input box, **replacing** the search button (I **need** the search button on desktop, instead of pressing `enter`)
2. Having the clear button on the left of search button (mis-clicking issue again)

**Comment by PikachuEXE** [comment]
> the button should disappear when the search input is empty

I think it looks strange.
But you all think this is ok I will push the change.

Screenshot 2021-09-03 at 11 07 39 AM
Screenshot 2021-09-03 at 11 07 41 AM

**Comment by jasonhenriquez** [comment]
I can't see the changes. Did you push the commits?

**Comment by PikachuEXE** [comment]
I excluded the change for "making the clear button when search input is empty"
Unless that change is blocking this PR from merging, I would like to leave the discussion for that design decision to another PR and make this PR merged first.

**Review by ChunkyProgrammer — DISMISSED** [review]
LGTM!

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]

**Comment by jasonhenriquez** [comment]
I only have one more suggestion: shorten the radius of the hover effect. I do not like that it obscures the mouse "typing" cursor when focus is in the search box.

**Comment by PikachuEXE** [comment]
Done
image

**Review by jasonhenriquez — APPROVED** [review]
LGTM!

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Review by ChunkyProgrammer — APPROVED** [review]

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
I think merge getting blocked by @peepopoggers requested changes.

**Review by peepopoggers — APPROVED** [review]

**Comment by peepopoggers** [comment]
@efb4f5ff-1298-471a-8973-3d47447115dc fixed :)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Yeah i think its the lint that's blocking it

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> Add an X button that will clear and select the search bar in one action/click.

@PikachuEXE 
I think we missed this part. At least in my build it doesn't select the search bar after clearing the text.

**Comment by jasonhenriquez** [comment]
This warrants a secondary PR that redirects focus and makes the button disappear after clicking.

**Comment by PikachuEXE** [comment]
#1669 created for auto focus
"makes the button disappear after clicking" < I am not sure if this is a good UX
I am skipping it for now

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Sounds logical to me to not show the button if there is no text to erase. Not sure about other implications.

**Comment by PikachuEXE** [comment]
Here is a preview for "makes the button disappear when input text box empty"
https://user-images.githubusercontent.com/1018543/132455351-b959d89e-28b2-438f-aa34-3a698f1ce0db.mov

**Comment by peepopoggers** [comment]
As i said before im still not a fan of the icon and placement but i guess thats preference ;/

**Comment by jasonhenriquez** [comment]
> Here is a preview for "makes the button disappear when input text box empty"
> https://user-images.githubusercontent.com/1018543/132455351-b959d89e-28b2-438f-aa34-3a698f1ce0db.mov

I think this would look nice if it was slower / smoother. Maybe with a css animation.

**Comment by PikachuEXE** [comment]
@jasonhenriquez 
Would you mind describing the css animation you are looking for?

**Comment by jasonhenriquez** [comment]
A fade out with a very brief timer.

**Comment by PikachuEXE** [comment]
@jasonhenriquez 
Please see if this is what you expect:
https://user-images.githubusercontent.com/1018543/132652935-5d4bc1f9-7066-409a-812e-090d3f1d49e2.mov

**Comment by peepopoggers** [comment]
looks pretty cool actually

**Comment by PikachuEXE** [comment]
Pushed all changes to #1669 
Lets continue the discussion there

**Pull request merged**
